compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -4779,6 +4779,152 @@ TR::Register *OMR::X86::TreeEvaluator::atomicorEvaluator(TR::Node *node, TR::Cod
    return NULL;
    }
 
+TR::Register *                                                                                                                                                             
+OMR::X86::TreeEvaluator::tstartEvaluator(TR::Node *node, TR::CodeGenerator *cg)                                                                                             
+   {                                                                                                                                                                       
+   /*         
+   .Lstart:
+      xbegin       .Lfallback
+      jmp          .LfallThrough
```

Comment:
The ideal way to do this is to have the fallback path be an out-of-line code section, where in you figure out why the transaction was aborted and branch to either the persistent or transient failure path. The `tstart` evaluator can then check whether the block representing the body of the transaction follows the current block or lives somewhere else and needs to be branched to.